R/step-subset-select.R

```diff
@@ -47,6 +46,34 @@ select.data.table <- function(.data, ...) {
   select(.data, ...)
 }
 
+dtplyr_tidyselect <- function(.data, ...,
+                              .env = caller_env(),
+                              .allow_rename = TRUE,
+                              .drop_groups = FALSE) {
+  dots <- enquos(...)
+  if (selection_uses_where(dots)) {
+    abort("The use of `where()` is not supported by dtplyr.")
```

Comment:
I think we want `abort(..., call = .env)` here so we get the name of the top-level function in the error e.g. 

```r
lazy_dt(tibble(y = 1)) %>% 
  select(where(y))
#> Error in `select()`:
#> ! The use of `where()` is not supported by dtplyr.
```

instead of

```r
lazy_dt(tibble(y = 1)) %>% 
  select(where(y))
#> Error in `dtplyr_tidyselect()` at dtp/R/step-subset-select.R:20:2:
#> ! The use of `where()` is not supported by dtplyr.
```

matching dplyr

```r
tibble(y = 1) %>% 
  select(where(fakefun))
#> Error in `select()`:
#> ! object 'fakefun' not found
```

Reply:
Hmm, looks like `.env` isn't passed around the right way for this to work in `mutate(across`, because the change I suggested above results in a blank first line in the code below. Not sure if it's worth trying to fix our errors in this PR. It would be good to get `where()` to stop silently failing and the error is still unambiguous, this is more of a minor dplyr-consistency thing.

```r
mutate(dt, across(!where(is.character), ~ .x + 1))
#> Error:
#> ! The use of `where()` is not supported by dtplyr.
```

Reply:
I was able to fix this by passing `.call = call2(fn)` in `across_setup()`. So whether using `across()` or `if_any()`/`if_all()` we now have the proper error message.